resolwe_bio/processes/alignment/star.py

```diff
@@ -0,0 +1,704 @@
+                default=1000,
+                description="Maximum number of junctions for one read (including all "
+                "multi-mappers).",
+            )
+
+            collapsed_junctions = IntegerField(
+                label="Maximum number of collapsed junctions [--limitOutSJcollapsed]",
+                default=1000000,
+            )
+
+            inserted_junctions = IntegerField(
+                label="Maximum number of junction to be inserted [--limitSjdbInsertNsj]",
+                default=1000000,
+                description="Maximum number of junction to be inserted to the genome on the fly  "
+                "at the mapping stage, including those from annotations and those detected in the "
+                "1st step of the 2-pass run.",
+            )
+
+        annotation_options = GroupField(
+            AnnotationOptions, label="Annotation file options", hidden="!annotation"
+        )
+        detect_chimeric = GroupField(
+            ChimericReadsOptions, label="Chimeric and circular alignments"
+        )
+        t_coordinates = GroupField(
+            TranscriptOutputOptions, label="Transcript coordinates output"
+        )
+        filtering = GroupField(FilteringOptions, label="Output Filtering")
+        alignment = GroupField(AlignmentOptions, label="Alignment and Seeding")
+        two_pass_mapping = GroupField(TwoPassOptions, label="Two-pass mapping")
+        output_options = GroupField(OutputOptions, label="Output options")
+        limits = GroupField(Limits, label="Limits")
+
+    class Output:
+        """Output fields to process AlignmentStar."""
+
+        bam = FileField(label="Alignment file")
+        bai = FileField(label="BAM file index")
+        unmapped_1 = FileField(label="Unmapped reads (mate 1)", required=False)
+        unmapped_2 = FileField(label="Unmapped reads (mate 2)", required=False)
+        sj = FileField(label="Splice junctions")
+        chimeric = FileField(label="Chimeric alignments", required=False)
+        alignment_transcriptome = FileField(
+            label="Alignment (trancriptome coordinates)", required=False
+        )
+        stats = FileField(label="Statistics")
+        bigwig = FileField(label="BigWig file", required=False)
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+        log = FileField(label="STAR log file")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        species = [
+            "Caenorhabditis elegans",
+            "Cricetulus griseus",
+            "Dictyostelium discoideum",
+            "Dictyostelium purpureum",
+            "Drosophila melanogaster",
+            "Homo sapiens",
+            "Macaca mulatta",
+            "Mus musculus",
+            "Odocoileus virginianus texanus",
+            "Rattus norvegicus",
+            "Solanum tuberosum",
+        ]
+        try:
+            sample_species = inputs.reads.entity.descriptor["general.species"]
```

Comment:
Instead of creating a variable `inputs.reads.entity.descriptor["general.species"]` will be used to check and write to descriptor. I am reluctant to change the `outputs.species` to anything other than the species of the reference genome to avoid mismatches with gtf files downstream (e.g. featureCounts).